src/compiler/compile.ts

```diff
@@ -184,15 +185,55 @@ export class Compile {
     return text
   }
 
+  // We should find another way to ensure SourceRefs
+  // are in sync, this seems like a hack
+  private outdentSourceRefs(
+    { text, sourceMap }: { text: string; sourceMap: PromptlSourceRef[] }
+  ): PromptlSourceRef[] {
+    const indent = getCommonIndent(text)
+    let position = 0
+    text = text.split('\n').map((line) => {
+      const offset = line.length - line.slice(indent).length
+      line = line.slice(indent)
+      sourceMap = sourceMap.map((ref) => ({
+        ...ref,
+        start: ref.start >= position ? ref.start - offset : ref.start,
+        end: ref.end >= position ? ref.end - offset : ref.end,
+      }))
+      position += line.length + 1
+      return line
+    }).join('\n')
+
+    const offset = text.length - text.trimStart().length
+    text = text.trimStart()
+    sourceMap = sourceMap.map((ref) => ({
+      ...ref,
+      start: ref.start - offset,
+      end: ref.end - offset,
+    }))
+
+    text = text.trimEnd()
+    sourceMap = sourceMap.map((ref) => ({
+      ...ref,
+      start: Math.min(ref.start, text.length),
+      end: Math.min(ref.end, text.length),
+    }))
+ 
+    return sourceMap
+  }
+
   private groupStrayText(): void {
-    const text = this.popStrayText()
-    if (!text.text.trim().length) return
+    const stray = this.popStrayText()
+    const text = removeCommonIndent(stray.text)
+    const sourceMap = this.outdentSourceRefs(stray);
+
```

Comment:
Removing common indent from `stray.text` could not mess with the contents of `stray.text` before you run `outdentSourceRefs`?

A less hacky way would be to both perform the indent removal and source refs recalculation in the same method (?)

Reply:
If it works, it works though

Reply:
Good catch, it could be, but I think strings are immutable in javascript, I will change the order tho!!